security-privacy-questionnaire-bluetooth-rfcomm.md

```diff
@@ -0,0 +1,122 @@
+# Security and Privacy Questionnaire
+
+Questions pulled from the [Self-Review Questionaire on Security and Privacy](https://www.w3.org/TR/security-privacy-questionnaire/).
+
+## What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+
+This questionnaire is specifically about the "Bluetooth RFCOMM in Web Serial API" feature which enables applications to access serial interfaces on Bluetooth devices. See EXPLAINER_BLUETOOTH.md.
+
+Much of the same information is exposed for Bluetooth RFCOMM serial ports as for non-Bluetooth serial ports. Copied from security-privacy-questionnaire.md:
+
+> By allowing a site to communicate with a peripheral device it is exposed to any information that the device makes available through that connection. For example, a device may provide a command for checking the currently installed firmware version or reading the serial number, which must be assumed to be unique. The device may also have access to information about the user's environment. For example, it may include a temperature sensor.
+
+> No effort is made by this specification to limit the type of data that can be read from the device once the user has granted a site permission to communicate with it. The reason for this is that serial device communication is a bidirectional byte stream that is entirely opaque to the user agent.
+
+> The trade-off being made is that users benefit from being able to perform a task using a web application rather than a native application. Web applications do not require installation and so are ideal for tasks that are performed infrequently. They are also sandboxed away from both the underlying platform and each other. Access to a powerful capability such as communication with a connected serial device can be placed behind a permission request that only grants a site access to a particular device, rather than the ability to run arbitrary native code on the user's system and access to all capabilities granted by default on the native platform.
+
+Specific to the Bluetooth RFCOMM case, we will add a new `SerialPortInfo` dictionary member `bluetoothServiceClassId` which will expose the service class UUID of the connected RFCOMM service.
+It is necessary to expose this information so that applications can distinguish between two `SerialPort` instances created for different RFCOMM services on the same Bluetooth device.
+The service UUID must be exposed without modification because the value is decided by the manufacturer and applications expect to be able to match against the exact value.
+[Design principles](https://www.w3.org/TR/design-principles/#device-ids) suggests to use a random number in place of unique identifiers.
+This is not possible for service class UUIDs since the exact value of the UUID is known by the application and used to identify supported features on the device.
```

Comment:
You can add that similar to USB vendor and product IDs these are not truly unique identifiers as every instance of a particular device model is expected to have a the same service UUIDs available.